platooning_tactical_plugin/src/platooning_tactical_plugin.cpp

```diff
@@ -0,0 +1,617 @@
+
+    std::vector<double> actual_speeds = apply_speed_limits(ideal_speeds, distributed_speed_limits);
+
+    log::printDoublesPerLineWithPrefix("actual_speeds: ", actual_speeds);
+
+    log::printDoublesPerLineWithPrefix("yaw_values[i]: ", yaw_values);
+
+    for (int i = 0; i < yaw_values.size() - 1; i++)
+    {  // Drop last point
+
+      Eigen::Isometry2d point_in_map = curvePointInMapTF(discreet_curve.frame, sampling_points[i], yaw_values[i]);
+      Eigen::Rotation2Dd new_rot(point_in_map.rotation());
+      final_yaw_values.push_back(new_rot.smallestAngle());
+      all_sampling_points.push_back(point_in_map.translation());
+    }
+
+    final_actual_speeds.insert(final_actual_speeds.end(), actual_speeds.begin(), actual_speeds.end() - 1);
+
+    ROS_DEBUG("Appended to final");
+  }
+
+
+  ROS_DEBUG("Processed all curves");
+
+  if (all_sampling_points.size() == 0)
+  {
+    ROS_WARN_STREAM("No trajectory points could be generated");
+    return {};
+  }
+
+  log::printDoublesPerLineWithPrefix("final_actual_speeds[i]: ", final_actual_speeds);
+
+  log::printDoublesPerLineWithPrefix("final_yaw_values[i]: ", final_yaw_values);
+
+  // Find Lookahead Distance based on Velocity
+  double lookahead_distance = get_adaptive_lookahead(state.longitudinal_vel);
+
+  ROS_DEBUG_STREAM("Lookahead distance at current speed: " << lookahead_distance);
+
+  // Apply lookahead speeds
+  final_actual_speeds = get_lookahead_speed(all_sampling_points, final_actual_speeds, lookahead_distance);
+  
+  // Add current vehicle point to front of the trajectory
+  lanelet::BasicPoint2d cur_veh_point(state.X_pos_global, state.Y_pos_global);
+  all_sampling_points.insert(all_sampling_points.begin(),
+                             cur_veh_point);  // Add current vehicle position to front of sample points
+
+  final_actual_speeds.insert(final_actual_speeds.begin(), std::max(state.longitudinal_vel, config_.minimum_speed));
+
+  final_yaw_values.insert(final_yaw_values.begin(), state.orientation);
+
+  log::printDoublesPerLineWithPrefix("pre_smoot[i]: ", final_actual_speeds);
+  
+  // Compute points to local downtracks
+  std::vector<double> downtracks = carma_wm::geometry::compute_arc_lengths(all_sampling_points);
+
+  log::printDoublesPerLineWithPrefix("post_shift[i]: ", final_actual_speeds);
+  
+  // Apply accel limits
+  final_actual_speeds = trajectory_utils::apply_accel_limits_by_distance(downtracks, final_actual_speeds,
+                                                                         config_.max_accel, config_.max_accel);
+  log::printDoublesPerLineWithPrefix("postAccel[i]: ", final_actual_speeds);
+
+  
+  final_actual_speeds = smoothing::moving_average_filter(final_actual_speeds, config_.moving_average_window_size);
+  log::printDoublesPerLineWithPrefix("post_average[i]: ", final_actual_speeds);
+
+  for (auto& s : final_actual_speeds)  // Limit minimum speed. TODO how to handle stopping?
```

Comment:
Hmm, handling a stop normally would be covered by the Stop&Wait plugin, but in the case of platooning especially that may not be desired. Good catch, something for us to think about!

Reply:
Should probably write up an issue to document that this edge case isn't covered.